packages/protocol/contracts/governance/DowntimeSlasherIntervals.sol

```diff
@@ -0,0 +1,409 @@
+   * [startIntervals(0), endIntervals(n)]
+   * @param startIntervals List of blocks that starts a previously validated Interval.
+   * startIntervals[0] will be use as the startBlock of the SlashableDowntime
+   * @param endIntervals List of blocks that ends a previously validated Interval.
+   * @param startSignerIndex Index of the signer within the validator set as of the start block.
+   * @param endSignerIndex Index of the signer within the validator set as of the end block.
+   * @return True if the validator signature does not appear in any block within the window.
+   */
+  function wasDownForIntervals(
+    uint256[] memory startIntervals,
+    uint256[] memory endIntervals,
+    uint256 startSignerIndex,
+    uint256 endSignerIndex
+  ) public view returns (bool) {
+    require(startIntervals.length > 0, "Requires at least one Interval");
+    require(startIntervals[0] > 0, "startIntervals[0] should be bigger than 0");
+    require(
+      startIntervals.length == endIntervals.length,
+      "StartIntervals and EndIntervals must have the same length"
+    );
+    uint256 endBlock = getEndBlockForSlashing(startIntervals[0]);
+    uint256 epochSize = getEpochSize();
+    uint256 epochChangeBlock = epochNumberOfBlock(startIntervals[0], epochSize).mul(epochSize).add(
+      1
+    );
+    uint256 signerIndex = startSignerIndex;
+    uint256 lastIntervalEnd = startIntervals[0];
+    for (uint256 i = 0; i < startIntervals.length; i += 1) {
+      require(
+        intervalProofAlreadyCalculated(startIntervals[i], endIntervals[i]),
+        "Interval has not been calculated"
+      );
+      require(
+        lastIntervalEnd >= startIntervals[i].sub(1) && lastIntervalEnd <= endIntervals[i],
+        "At least one endInterval is not in te boundaries of the next Interval"
+      );
+      if (startIntervals[i] >= epochChangeBlock) {
+        signerIndex = endSignerIndex;
+      }
+      if (!wasDownUsingIntervalProof(startIntervals[i], endIntervals[i], signerIndex)) {
+        return false;
+      }
+      lastIntervalEnd = endIntervals[i];
+      // Could be covered with less intervals
+      if (lastIntervalEnd >= endBlock) {
+        return true;
+      }
+    }
+
+    require(
+      lastIntervalEnd >= endBlock,
+      "The Intervals are not covering the SlashableDowntime window"
+    );
+
+    return true;
+  }
+
+  /**
+   * @notice Returns the end block for the interval required for slashing.
+   */
+  function getEndBlockForSlashing(uint256 startBlock) internal view returns (uint256) {
+    return startBlock.add(slashableDowntime).sub(1);
+  }
+
+  /**
+   * @notice Add to the validator a new SlashedInterval if the validator was not already
+   * slashed by a previous interval that contains al least a block of the new interval.
+   * If the oncePerEpoch flag is set, won't allow two slashes in the same epoch for the same
+   * validator 
+   */
+  function addNewSlashIntervalToValidator(address validator, uint256 startBlock) internal {
```

Comment:
Changed yes, but not removed. You should also check if the crossing epoch slashes have some blocks in common
Let's say the validator1 was down from the `startBlockEpochN  -  10` (in epochN-1 to the `startBlockEpochN  +  100`.
If the new slash it's from `startBlochEpochN  + 20` it won't be a valid one, because they share blocks.
The thing is that if you have a user that was down 8 hours + 1 minute, and only part of that minute was in a previous epoch, you will be able to slash it twice.